### ISSUE TITLE
java.lang.IllegalArgumentException: Cross module dependency resolution failed

### ISSUE THREAD

**Issue by DennisTsar** [JetBrains/compose-multiplatform#3424]
**Describe the bug**
When upgrading to compose 1.4.3 and kotlin 1.9.0, the code 

```kt
internal class MyClass {
    @Composable
    fun test() {
        Div {
            Text("Error")
        }
    }
}

internal val MyCompositionLocal = staticCompositionLocalOf<MyClass?> { null }
```

fails to compile:
`e: java.lang.IllegalArgumentException: Cross module dependency resolution failed due to signature 'kotlinx.dom/clear|6741251430650539538[0]' redefinition`

<details>
  <summary>Stack trace</summary>

  ```
e: java.lang.IllegalArgumentException: Cross module dependency resolution failed due to signature 'kotlinx.dom/clear|6741251430650539538[0]' redefinition
        at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.JsIrModuleCrossModuleReferenceBuilder.buildCrossModuleRefs(JsIrProgramFragment.kt:168)
        at org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.CrossModuleDependenciesResolver.resolveCrossModuleDependencies(JsIrProgramFragment.kt:131)
        at org.jetbrains.kotlin.ir.backend.js.ic.JsExecutableProducer.buildMultiModuleExecutable(JsExecutableProducer.kt:56)
        at org.jetbrains.kotlin.ir.backend.js.ic.JsExecutableProducer.buildExecutable(JsExecutableProducer.kt:30)
        at org.jetbrains.kotlin.cli.js.K2JsIrCompiler.doExecute(K2JsIrCompiler.kt:309)
        at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:181)
        at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:72)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:100)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:46)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
        at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1497)
        at jdk.internal.reflect.GeneratedMethodAccessor31.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:359)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:562)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:796)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:677)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:676)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
  ```
</details>

**Affected platforms**
Select one of the platforms below:
- Web (K/JS) - HTML library

**Versions**
- Kotlin version*: 1.9.0
- Compose Multiplatform version*: 1.4.3, 1.5.0-beta01

**To Reproduce**
A minimal reproducing project is available at https://github.com/DennisTsar/compose-bug-repro. Running `./gradlew site:compileDevelopmentExecutableKotlinJs` will cause the failure.

**Expected behavior**
The project compiles successfully.

**Additional context**
- The code works with compose 1.4.1 with kotlin 1.8.20 and compose 1.4.3 with kotlin 1.8.22
- Production compilation with `./gradlew site:compileProductionExecutableKotlinJs` finishes successfully
- Moving `implementation(compose.runtime)` from under `commonMain` to under `jsMain` fixes the issue

**Comment by eymar** [comment]
Thanks!
On a first glance, it seems like an issue in kotlin (perhaps specific only to kotlin/js gradle plugin).

I tried a newer kotlin and a dev compose compiler plugin:

```
Index: gradle/libs.versions.toml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
--- a/gradle/libs.versions.toml	(revision b9322a50992500b82c730b0d339211c32af80d76)
+++ b/gradle/libs.versions.toml	(date 1690560470192)
@@ -1,6 +1,7 @@
 [versions]
 jetbrains-compose = "1.4.3"
-kotlin = "1.9.0"
+kotlin = "1.9.20-dev-6473"
+kotlinCompilerPluginVersion="0.0.0-1.9.20-dev-6336"
 
 [plugins]
 jetbrains-compose = { id = "org.jetbrains.compose", version.ref = "jetbrains-compose" }
Index: build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle.kts b/build.gradle.kts
--- a/build.gradle.kts	(revision b9322a50992500b82c730b0d339211c32af80d76)
+++ b/build.gradle.kts	(date 1690560448264)
@@ -7,5 +7,6 @@
     repositories {
         mavenCentral()
         maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
+        maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/")
     }
 }
\ No newline at end of file
Index: settings.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/settings.gradle.kts b/settings.gradle.kts
--- a/settings.gradle.kts	(revision b9322a50992500b82c730b0d339211c32af80d76)
+++ b/settings.gradle.kts	(date 1690560160862)
@@ -2,6 +2,7 @@
     repositories {
         gradlePluginPortal()
         maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
+        maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/")
     }
 }
 
Index: site/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/site/build.gradle.kts b/site/build.gradle.kts
--- a/site/build.gradle.kts	(revision b9322a50992500b82c730b0d339211c32af80d76)
+++ b/site/build.gradle.kts	(date 1690560160854)
@@ -24,4 +24,8 @@
             }
         }
     }
+}
+
+compose {
+    kotlinCompilerPlugin.set(libs.versions.kotlinCompilerPluginVersion)
 }
\ No newline at end of file
```

And the project starts compiling. Afaik the dev compose compiler plugin didn't introduce any new changes comparing to 1.9.0-compatible version beside some 1.9.20-dev related changes.

And given the fact that commenting the dependency in commonMain and uncommenting it in jsMain makes it work, suggests that it's likely something in kotlin.
I'll try to find a commit in kotlin that fixes it.

**Comment by hfhbd** [comment]
Hey, I got the same problem and opened a KT-issue: https://youtrack.jetbrains.com/issue/KT-60852/Kotlin-JSIR-IllegalArgumentException-Cross-module-dependency-resolution-failed-when-updating-to-Kotlin-1.9.0 redirecting to this issue.

**Comment by DennisTsar** [comment]
Fixed in compose 1.5.1 with kotlin 1.9.10

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3424">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
